atlp-rwanda/e-comm-team-brogrammers-bn
# #184581329  Receiving email verification link upon…

#### What does this PR do?
- creating account for a new user and validate his/her information
- set the user id to `uuid` format
- testing the user signup routes
- adding documentation of the user signup route
- Receiving email verification link upon successfully register

#### Description of Task to be completed?
user must enter the data such as
- username
- email
- password
- gender

then it validates those information
then check if there isn't other account with that email address
and add it in the database 
after return the message that user has been created and also receive email verification link on his/her email  to verify email
endpoints:
- /users/signup

#### How should this be manually tested?
after cloning the repo,
- in terminal add `npm install`
- add `.env` in your repo and assign the variables as there are in `.env.example` file
- run in terminal `npm run migrate:all` to setup the database
- run in terminal `npm start` to start the project
- Change SWAGGER_SERVER=https://ecommerceee.onrender.com/
- in your browser add `https://ecommerceee.onrender.com/api-docs/` for documentantion
- Then go to signup route
- upon successfully registration you receive  a verification link and click on it to verify email

#### Any background context you want to provide?
n/a

#### What are the relevant pivotal tracker/Trello stories?
https://www.pivotaltracker.com/n/projects/2632830/stories/184581329

#### Screenshots (if appropriate)
<img width="889" alt="Screenshot 2023-03-12 at 17 02 05" src="https://user-images.githubusercontent.com/56297754/224558621-f6ee3c25-474e-4bda-8484-906c351486b4.png">
<img width="1418" alt="Screenshot 2023-03-12 at 18 37 00" src="https://user-images.githubusercontent.com/56297754/224558802-dfc795ea-77f8-45c8-9b04-87a8d98cc968.png">
<img width="1131" alt="Screenshot 2023-03-12 at 18 37 27" src="https://user-images.githubusercontent.com/56297754/224558810-d55a0850-e74f-4117-997e-ed90d4430e1e.png">
<img width="1440" alt="Screenshot 2023-03-12 at 18 38 43" src="https://user-images.githubusercontent.com/56297754/224558880-fbcab25b-a74e-41f5-8d0d-76cc9346dedd.png">
<img width="1286" alt="Screenshot 2023-03-12 at 18 50 05" src="https://user-images.githubusercontent.com/56297754/224559673-350c09b6-735f-492c-9220-655dd59f201d.png">




#### Questions:
n/a
